Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -685,25 +686,26 @@ s32 WiiSockMan::DeleteSocket(s32 s)
 
 void WiiSockMan::Update()
 {
-  s32 nfds = 0;
-  fd_set read_fds, write_fds, except_fds;
-  struct timeval t = {0, 0};
-  FD_ZERO(&read_fds);
-  FD_ZERO(&write_fds);
-  FD_ZERO(&except_fds);
+  constexpr int write_event = (POLLOUT | POLLWRBAND);
+  constexpr int read_event = (POLLIN | POLLRDNORM | POLLRDBAND | POLLPRI);
+  constexpr int error_event = (POLLHUP | POLLERR);
 
   auto socket_iter = WiiSockets.begin();
   auto end_socks = WiiSockets.end();
+  ufds.clear();
+  ufds.reserve(WiiSockets.size());
 
   while (socket_iter != end_socks)
   {
     WiiSocket& sock = socket_iter->second;
     if (sock.IsValid())
     {
-      FD_SET(sock.fd, &read_fds);
-      FD_SET(sock.fd, &write_fds);
-      FD_SET(sock.fd, &except_fds);
-      nfds = std::max(nfds, sock.fd + 1);
+#ifdef _WIN32
+      // On Win32, WSAPoll only accepts SOCKET types as file descriptors
+      ufds.push_back({static_cast<SOCKET>(sock.fd), read_event | write_event, 0});
```

Comment:
You will need to remove Windows' unsupported events here with `UNSUPPORTED_WSAPOLL`. Feel free to move it to `Socket.h`.